doc/autogen.py

```diff
@@ -108,6 +108,8 @@ def python_doc_stages(src_dir, dst_dir, list_file):
     # something should be really wrong if this asserts
     assert len(all_classes) == len(all_files)
     gen = Template("""
+.. _dsp_stages_list:
```

Comment:
Does Sphinx need an empty line above the anchor?  I'm unsure, but I know that it does need such lines in some cases.